### PR TITLE
[BUG]: sql nested and wildcard

### LINKED ISSUE #2935: Reproducing with_column functionality in SQL
### Describe the bug

I would like to express `df = df.with_column("x", df["A"] + 1)` in SQL

I am not able to make this work for 2 reasons:

image

image

### To Reproduce

```
            df = daft.from_pydict({
                "A": [1, 2, 3, 4],
                "B": [1.5, 2.5, 3.5, 4.5],
                "C": [True, True, False, False],
                "D": [None, None, None, None],
            })

           daft.sql("SELECT *, (A + 1) AS try_this FROM df").show(5)
```

### Expected behavior

_No response_

### Component(s)

SQL

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by universalmind303**
closes #2935

### CODE REVIEW

File: src/daft-sql/src/lib.rs [hunk 1]
```diff
@@ -261,6 +261,12 @@ mod tests {
                 JoinType::Inner,
                 None,
             )?
+            .select(vec![
```
Comment (universalmind303) [inline src/daft-sql/src/lib.rs:264]: wildcards now always expand to a `.select(...)`, which shouldnt be an issue as the optimizer should result in the same optimized plan

Comment (jaychia) [reply src/daft-sql/src/lib.rs:264]: How would this interact with conflicting names BTW?

I wonder what the expected behavior should be if we do `SELECT *, (tbl.A + 1) AS A FROM tbl` and want to "override" the column "A"

**Review by jaychia — APPROVED** [review]

File: src/daft-sql/src/lib.rs (hunk 1 above)
Comment (jaychia) [reply src/daft-sql/src/lib.rs:264]: Also we do have some handling of `col("*")` that @Vince7778 implemented, wonder if we could leverage the same machinery: https://github.com/Eventual-Inc/Daft/pull/2629

Comment (universalmind303) [reply src/daft-sql/src/lib.rs:264]: oh yeah! How did I forget about our `col('*')` support. Will update!

**Comment by universalmind303** [comment]
> How would this interact with conflicting names BTW?
I wonder what the expected behavior should be if we do SELECT *, (tbl.A + 1) AS A FROM tbl and want to "override" the column "A"

This should work the same as our dataframe select `df.select(col('*'), col('A'))` which currently returns an error.

**Pull request merged**